django/contrib/contenttypes/fields.py

```diff
@@ -25,6 +27,18 @@
 from django.utils.functional import cached_property
 
 
+def serialize_pk(pk):
+    if isinstance(pk, (tuple, list)):
+        pk = json.dumps(pk, cls=DjangoJSONEncoder)
+    return pk
+
+
+def deserialize_pk(pk):
+    if pk.startswith("["):
+        pk = json.loads(pk)
+    return pk
```

Comment:
I don't think we can use this approach. It'll be rare, but it's completely valid to have a string field as primary key with a value that's a valid json list.

Reply:
Yes, you're right

Reply:
We will need to check if a composite primary key is _expected_ ... this was just a proof of concept, I wanted to check if the approach is reasonable first. I can adjust this once composite primary keys are merged.